File: calculator/src/main/java/com/turlir/calculator/interpreter/NotationInterpreter.java
package com.turlir.calculator.interpreter;


import java.math.BigDecimal;

/**
 * Вычисляет выражение, записанное в определенной нотации
 */
public interface NotationInterpreter {

    /**
     *
     * @return промужеточный результат
     */
    BigDecimal poolDigit();

    /**
     * Записывает промежуточный результат
     * @param value результат
     */
    void pushDigit(BigDecimal value);

}

File: calculator/src/main/java/com/turlir/calculator/interpreter/NotationInterpreter.kt
package com.turlir.calculator.interpreter


import java.math.BigDecimal

/**
 * Вычисляет выражение, записанное в определенной нотации
 */
interface NotationInterpreter {

    /**
     *
     * @return промужеточный результат
     */
    fun poolDigit(): BigDecimal

    /**
     * Записывает промежуточный результат
     * @param value результат
     */
    fun pushDigit(value: BigDecimal)

}
